drivers/spi/spi_ll_stm32.c

```diff
@@ -106,11 +106,9 @@ static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 /* Shift a SPI frame as slave. */
 static void spi_stm32_shift_s(SPI_TypeDef *spi, struct spi_stm32_data *data)
 {
-	u16_t tx_frame;
-	u16_t rx_frame;
+	if (LL_SPI_IsActiveFlag_TXE(spi) && spi_context_tx_on(&data->ctx)) {
+		u16_t tx_frame = spi_stm32_next_tx(data);
 
-	tx_frame = spi_stm32_next_tx(data);
-	if (LL_SPI_IsActiveFlag_TXE(spi)) {
 		if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
 			LL_SPI_TransmitData8(spi, tx_frame);
 			/* The update is ignored if TX is off. */
```

Comment:
This comment and its duplicate below should be removed now, since this change checks that TX is on earlier.

Reply:
ok